src/data_structures/suffix_array.rs

```diff
@@ -19,12 +20,79 @@ use vec_map::VecMap;
 
 use alphabets::{Alphabet, RankTransform};
 use data_structures::smallints::SmallInts;
+use data_structures::bwt::{BWT, Less, Occ};
 
+pub trait SuffixArray {
+    fn get(&self, index: usize) -> Option<usize>;
+    fn range(&self, range: Range<usize>) -> Option<Vec<usize>>;
```

Comment:
How would you prefer `Interval::occ` worked, then? Should I just get rid of it, or...?